tests/profiler/test_rule_state.py

```diff
@@ -12,38 +16,128 @@
+    )
+
+
+@pytest.fixture
+def single_syllable_parameter_container():
+    return ParameterContainer(parameters={"mean": 5.0}, details=None, descendants=None)
+
+
+@pytest.fixture
+def multi_syllable_parameter_container():
+    """
+    $parameter.date_strings.yyyy_mm_dd_hh_mm_ss_tz_date_format
+    $parameter.date_strings.yyyy_mm_dd_date_format
+    $parameter.date_strings.mm_yyyy_dd_hh_mm_ss_tz_date_format
+    $parameter.date_strings.mm_yyyy_dd_date_format
+    $parameter.date_strings.tolerances.max_abs_error_time_milliseconds
+    $parameter.date_strings.tolerances.max_num_conversion_attempts
+    $parameter.tolerances.mostly
+    """
+    tolerances_parameter_container: ParameterContainer = ParameterContainer(
+        parameters={
+            "mostly": 9.3e-1,
+        },
+        details=None,
+        descendants=None,
+    )
+    date_strings_tolerances_parameter_container: ParameterContainer = (
+        ParameterContainer(
+            parameters={
+                "max_abs_error_time_milliseconds": 100,
+                "max_num_conversion_attempts": 5,
+            },
+            details=None,
+            descendants=None,
+        )
+    )
+    date_strings_parameter_container: ParameterContainer = ParameterContainer(
+        parameters={
+            "yyyy_mm_dd_hh_mm_ss_tz_date_format": "%Y-%m-%d %H:%M:%S %Z",
+            "yyyy_mm_dd_date_format": "%Y-%m-%d",
+            "mm_yyyy_dd_hh_mm_ss_tz_date_format": "%m-%Y-%d %H:%M:%S %Z",
+            "mm_yyyy_dd_date_format": "%m-%Y-%d",
+        },
+        details={
+            "confidence": {
+                "yyyy_mm_dd_hh_mm_ss_tz_date_format": 1.0,
+                "yyyy_mm_dd_date_format": 1.0,
+                "mm_yyyy_dd_hh_mm_ss_tz_date_format": 8.0e-1,
+                "mm_yyyy_dd_date_format": 8.0e-1,
+            }
+        },
+        descendants={
+            "tolerances": date_strings_tolerances_parameter_container,
+        },
+    )
+    parameter_syllable_level_parameter_container: ParameterContainer = (
+        ParameterContainer(
+            parameters=None,
+            descendants={
+                "date_strings": date_strings_parameter_container,
+                "tolerances": tolerances_parameter_container,
+            },
+        )
+    )
+    root_parameter_container: ParameterContainer = ParameterContainer(
+        parameters=None,
+        descendants={
+            "parameter": parameter_syllable_level_parameter_container,
+        },
+    )
+    return root_parameter_container
+
+
+# noinspection PyPep8Naming
 @pytest.fixture
-def simple_rule_state():
+def rule_state_with_no_parameters(
+    column_Age_storage_type_domain, column_Date_storage_type_domain
+):
     """Simple rule_state with one domain, currently set to active"""
     return RuleState(
-        active_domain=Domain(
-            domain_kwargs={"column": "Age"}, domain_type=StorageDomainTypes.COLUMN
-        ),
+        active_domain=column_Age_storage_type_domain,
```

Comment:
Where did we leave off on the discussion of `active_domain`? Was this something that we decided to keep for now but revisit in the near future?

Reply:
@talagluck We are leaving it in for now, because while we pondered about finding a better name for `RuleState`, we have not done so; furthermore, we could not find justification for moving any of its current fields to other classes.  While we may do so in the future, we have decided to keep it intact for now.